service/history/hsm/tree.go

```diff
@@ -415,6 +464,46 @@ func (n *Node) CompareState(incomingNode *Node) (int, error) {
 // Sync updates the state of the current node to that of the incoming node.
 // Meant to be used by the framework, **not** by components.
 func (n *Node) Sync(incomingNode *Node) error {
```

Comment:
Let's revert these changes. This isn't how replication works. Leave replication for a follow-up PR.

Reply:
Done